client/src/RatingsReviews/RatingsReviews.jsx

```diff
@@ -1,251 +1,235 @@
       .then((response) => {
-        // console.log('successfully fetched ratings');
-        // console.log(response.data);
         const sum = getTotalRatings(response.data.ratings)[0];
         const totalRatings = getTotalRatings(response.data.ratings)[1];
-        this.setState(
-          {
-            meta: response.data,
-            average: (sum / totalRatings).toFixed(2),
-            ratings: response.data.ratings,
-            totalRatings,
-            isLoaded: true,
-          },
-        );
+        setMeta(response.data);
+        setAverage((sum / totalRatings).toFixed(1));
+        setRatings(response.data.ratings);
+        setTotalRatings(totalRatings);
+        setIsLoaded(true);
       })
       .catch((err) => console.log('error fetching ratings', err));
-  }
+  };
 
-  setRatingFilter(rating) {
-    const temp = this.state.ratingFilter;
-    if (!temp[rating]) {
-      temp[rating] = true;
-    } else {
-      temp[rating] = false;
-    }
-    // if there is not a single true in rating filter,
-    // set filteredbyrating to false.
-    if (Object.values(temp).indexOf(true) !== -1) {
-      this.setState({ filteredByRating: true, ratingFilter: temp }, () => {
-        this.getByRating();
-      });
+  useEffect(() => {
+    getReviews();
+    getRatings();
+  }, []);
+
+  const setRatingsFilter = (rating) => {
+    const index = ratingFilter.indexOf(rating);
+    if (index === -1) {
+      setRatingFilter((a) => [...a, rating]);
     } else {
-      this.setState({ filteredByRating: false, ratingFilter: temp, filtered: [] }, () => {
-      });
+      const temp = ratingFilter;
+      temp.splice(index, 1);
+      setRatingFilter(() => [...temp]);
     }
-  }
+  };
 
-  getByRating() {
+  const getByRating = () => {
     // set temp as current list of reviews,
     // filter temp to fit ratings filter,
     // set state reviews to be temp.
-    const temp = this.state.reviews;
-    const obj = this.state.ratingFilter;
-    const filtered = temp.filter((review) => {
-      if (obj[`${review.rating}`]) {
+    const temp = reviews;
+    const arr = ratingFilter;
+    const filteredReviews = temp.filter((review) => {
+      // if review.rating is found in array, return that review.
+      const index = arr.indexOf(`${review.rating}`);
+      if (index !== -1) {
         return review;
       }
     });
-    this.setState({ filtered });
-  }
-
-  render() {
-    return (
-      <StyledMain>
-        <StyledTitle>
-          <div>
-            RATINGS & REVIEWS
-          </div>
-        </StyledTitle>
-        <StyledInner>
-          <Ratings
-            meta={this.state.meta}
-            isLoaded={this.state.isLoaded}
-            average={this.state.average}
-            totalRatings={this.state.totalRatings}
-            setRatingFilter={this.setRatingFilter}
-            ratingFilter={this.state.ratingFilter}
+    setFiltered(filteredReviews);
+  };
+
+  useEffect(() => {
+    console.log('ratingfilter is changing');
```

Comment:
when merging into the main, i think we should delete all console.logs other than error stuff!